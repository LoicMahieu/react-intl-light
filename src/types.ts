export type PrimitiveType = string | number | boolean | null | undefined | Date;

export interface MessageDescriptor {
  id: string;
  description?: string | object;
  defaultMessage?: string;
}

export interface IntlConfig {
  locale: string;
  timeZone?: string;
  // formats: CustomFormats;
  textComponent?: React.ComponentType | keyof React.ReactHTML;
  messages: Record<
    string,
    string
  > /* | Record<string, MessageFormatElement[]> */;
  defaultLocale: string;
  // defaultFormats: CustomFormats;
  onError(err: string): void;
}

export interface IntlShape extends IntlConfig, IntlFormatters {
  // formatters: Formatters;
}

export interface IntlFormatters {
  // formatDate(
  //   value: Parameters<Intl.DateTimeFormat['format']>[0] | string,
  //   opts?: FormatDateOptions
  // ): string;
  // formatTime(
  //   value: Parameters<Intl.DateTimeFormat['format']>[0] | string,
  //   opts?: FormatDateOptions
  // ): string;
  // formatDateToParts(
  //   value: Parameters<Intl.DateTimeFormat['format']>[0] | string,
  //   opts?: FormatDateOptions
  // ): Intl.DateTimeFormatPart[];
  // formatTimeToParts(
  //   value: Parameters<Intl.DateTimeFormat['format']>[0] | string,
  //   opts?: FormatDateOptions
  // ): Intl.DateTimeFormatPart[];
  // formatRelativeTime(
  //   value: Parameters<IntlRelativeTimeFormat['format']>[0],
  //   unit?: Parameters<IntlRelativeTimeFormat['format']>[1],
  //   opts?: FormatRelativeTimeOptions
  // ): string;
  // formatNumber(
  //   value: Parameters<Intl.NumberFormat['format']>[0],
  //   opts?: FormatNumberOptions
  // ): string;
  // formatNumberToParts(
  //   value: Parameters<Intl.NumberFormat['format']>[0],
  //   opts?: FormatNumberOptions
  // ): Intl.NumberFormatPart[];
  // formatPlural(
  //   value: Parameters<Intl.PluralRules['select']>[0],
  //   opts?: FormatPluralOptions
  // ): ReturnType<Intl.PluralRules['select']>;
  formatMessage(
    descriptor: MessageDescriptor,
    values?: Record<string, PrimitiveType>
  ): string;
  formatMessage(
    descriptor: MessageDescriptor,
    values?: Record<string, PrimitiveType>
  ): string | React.ReactNodeArray;
  formatHTMLMessage(
    descriptor: MessageDescriptor,
    values?: Record<string, PrimitiveType>
  ): string;
}
