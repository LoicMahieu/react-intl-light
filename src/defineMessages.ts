import { MessageDescriptor } from './types';

/*
 * Copyright 2015, Yahoo Inc.
 * Copyrights licensed under the New BSD License.
 * See the accompanying LICENSE file for terms.
 */

type Messages<Names extends keyof any = string> = {
  [key in Names]: MessageDescriptor;
};

export default function defineMessages<Names extends keyof any>(
  messageDescriptors: Messages<Names>
) {
  // This simply returns what's passed-in because it's meant to be a hook for
  // babel-plugin-react-intl.
  return messageDescriptors;
}
