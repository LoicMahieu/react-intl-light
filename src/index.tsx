export * from './defineMessages';
export * from './components/FormattedHTMLMessage';
export * from './components/FormattedMessage';
export * from './components/provider';
export * from './components/useIntl';
export * from './types';
