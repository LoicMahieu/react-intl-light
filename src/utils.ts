import React from 'react';
import { IntlConfig } from './types';
// Since rollup cannot deal with namespace being a function,
// this is to interop with TypeScript since `invariant`
// does not export a default
// https://github.com/rollup/rollup/issues/1267
import * as invariant_ from 'invariant';
const invariant: typeof invariant_.default =
  (invariant_ as any).default || invariant_;

export const DEFAULT_INTL_CONFIG: Pick<
  IntlConfig,
  // | 'formats'
  | 'messages'
  | 'timeZone'
  | 'textComponent'
  | 'defaultLocale'
  // | 'defaultFormats'
  | 'onError'
> = {
  // formats: {},
  messages: {},
  timeZone: undefined,
  textComponent: React.Fragment,

  defaultLocale: 'en',
  // defaultFormats: {},

  onError: defaultErrorHandler,
};

export function defaultErrorHandler(error: string) {
  if (process.env.NODE_ENV !== 'production') {
    console.error(error);
  }
}

export function createError(message: string, exception?: Error) {
  const eMsg = exception ? `\n${exception.stack}` : '';
  return `[React Intl] ${message}${eMsg}`;
}

export function invariantIntlContext(intl?: any) {
  invariant(
    intl,
    '[React Intl] Could not find required `intl` object. ' +
      '<IntlProvider> needs to exist in the component ancestry.'
  );
}
