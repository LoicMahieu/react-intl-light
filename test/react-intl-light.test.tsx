import * as React from 'react';
import { IntlProvider, FormattedMessage, FormattedHTMLMessage } from '../src';
import renderer from 'react-test-renderer';
import defineMessages from '../src/defineMessages';

describe('react-intl-light', () => {
  describe('FormattedMessage', () => {
    it('without any message', () => {
      const component = renderer.create(
        <IntlProvider locale="fr" defaultLocale="fr">
          <h1>
            <FormattedMessage id="foo" defaultMessage="bar" />
          </h1>
          <h1>
            <FormattedMessage id="foo" />
          </h1>
          <h1>
            <FormattedMessage id="foo2" defaultMessage="bar2">
              {message => <span>The message: {message}</span>}
            </FormattedMessage>
          </h1>
          <h1>
            <FormattedHTMLMessage
              id="foo3"
              defaultMessage="<span>hello world</span>"
            />
          </h1>
        </IntlProvider>
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
    it('without any message but defautlLocale is different', () => {
      const component = renderer.create(
        <IntlProvider locale="fr" defaultLocale="en">
          <h1>
            <FormattedMessage id="foo" defaultMessage="bar" />
          </h1>
          <h1>
            <FormattedMessage id="foo" />
          </h1>
          <h1>
            <FormattedMessage id="foo2" defaultMessage="bar2">
              {message => <span>The message: {message}</span>}
            </FormattedMessage>
          </h1>
          <h1>
            <FormattedHTMLMessage
              id="foo3"
              defaultMessage="<span>hello world</span>"
            />
          </h1>
        </IntlProvider>
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
    it('with some messages', () => {
      const component = renderer.create(
        <IntlProvider
          locale="fr"
          messages={{
            foo: 'bim',
            foo2: 'bim2',
            foo3: '<span>bim3</span>',
          }}
        >
          <h1>
            <FormattedMessage id="foo" defaultMessage="bar" />
          </h1>
          <h1>
            <FormattedMessage id="foo2" defaultMessage="bar2">
              {message => <span>The message: {message}</span>}
            </FormattedMessage>
          </h1>
          <h1>
            <FormattedHTMLMessage
              id="foo3"
              defaultMessage="<span>hello world</span>"
            />
          </h1>
        </IntlProvider>
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe('defineMessages', () => {
    it('works', () => {
      const messages = defineMessages({
        foo: {
          id: 'foo',
          defaultMessage: 'bar',
        },
      });

      expect(messages).toMatchSnapshot();
    });
  });
});
