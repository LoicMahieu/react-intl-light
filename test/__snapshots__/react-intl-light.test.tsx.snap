// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-intl-light FormattedMessage with some messages 1`] = `
Array [
  <h1>
    bim
  </h1>,
  <h1>
    <span>
      The message: 
      bim2
    </span>
  </h1>,
  <h1>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span>bim3</span>",
        }
      }
    />
  </h1>,
]
`;

exports[`react-intl-light FormattedMessage without any message 1`] = `
Array [
  <h1>
    bar
  </h1>,
  <h1 />,
  <h1>
    <span>
      The message: 
      bar2
    </span>
  </h1>,
  <h1>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span>hello world</span>",
        }
      }
    />
  </h1>,
]
`;

exports[`react-intl-light FormattedMessage without any message but defautlLocale is different 1`] = `
Array [
  <h1>
    bar
  </h1>,
  <h1 />,
  <h1>
    <span>
      The message: 
      bar2
    </span>
  </h1>,
  <h1>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span>hello world</span>",
        }
      }
    />
  </h1>,
]
`;

exports[`react-intl-light defineMessages works 1`] = `
Object {
  "foo": Object {
    "defaultMessage": "bar",
    "id": "foo",
  },
}
`;
